nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/field/TestObjectLocalDateTimeFieldConverter.java

```diff
@@ -0,0 +1,99 @@
+ * limitations under the License.
+ */
+
+package org.apache.nifi.serialization.record.field;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestObjectLocalDateTimeFieldConverter {
+
+    @Test
+    public void testConvertTimestampMillis() {
+        final ObjectLocalDateTimeFieldConverter converter = new ObjectLocalDateTimeFieldConverter();
+        final long now = System.currentTimeMillis();
+        final Instant instant = Instant.ofEpochMilli(now);
+        final LocalDateTime expected = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+
+        final LocalDateTime result = converter.convertField(now, Optional.empty(), "test");
+        assertEquals(expected, result);
+        assertEquals(now, result.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli());
+    }
+
+    @Test
+    public void testConvertTimestampMicros() {
+        final ObjectLocalDateTimeFieldConverter converter = new ObjectLocalDateTimeFieldConverter();
+        final long now = System.currentTimeMillis();
+        final long withMicros =  now * 1000 + 567;
+
+        final LocalDateTime result = converter.convertField(withMicros, Optional.empty(), "test");
+        assertEquals(now, result.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli());
+
+        final Instant resultInstant = result.atZone(ZoneId.systemDefault()).toInstant();
+        final long expectedNanos = now % 1000 * 1_000_000 + 567_000;
+        assertEquals(expectedNanos, resultInstant.getNano());
+    }
+
+    @Test
+    public void testDoubleAsEpochSeconds() {
+        final ObjectLocalDateTimeFieldConverter converter = new ObjectLocalDateTimeFieldConverter();
+        final long millis = System.currentTimeMillis();
+        final double seconds = millis / 1000.0;
+        final double withMicros =  seconds + 0.000567;
+
+        final LocalDateTime result = converter.convertField(withMicros, Optional.empty(), "test");
+        assertEquals(LocalDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneId.systemDefault()).toLocalDate(), result.toLocalDate());
+        final double expectedNanos = (withMicros - (long) seconds) * 1_000_000_000;
+        assertEquals(expectedNanos, result.getNano(), 1000D);
+    }
+
+    @Test
+    public void testDoubleAsEpochSecondsAsString() {
+        final ObjectLocalDateTimeFieldConverter converter = new ObjectLocalDateTimeFieldConverter();
+        final long millis = System.currentTimeMillis();
+        final double seconds = millis / 1000.0;
+        final double withMicros =  seconds + 0.000567;
+
+        final LocalDateTime result = converter.convertField(Double.toString(withMicros), Optional.empty(), "test");
+        assertEquals(LocalDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneId.systemDefault()).toLocalDate(), result.toLocalDate());
```

Comment:
This comparison is a bit difficult to follow, but it appears to be comparing only the `LocalDate` portion of `LocalDateTime`. Is there a reason for this comparison as opposed to the nanosecond comparison below?